src/zap/zapinfo.cpp

```diff
@@ -2086,6 +2104,30 @@ void ZapInfo::GetProfilingHandle(BOOL                      *pbHookFunction,
     *pbIndirectedHandles = TRUE;
 }
 
+DWORD FilterMethodAttribsForIsSupported(DWORD attribs, CORINFO_METHOD_HANDLE ftn, ICorDynamicInfo* pJitInfo)
+{
+#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
+    if (attribs & CORINFO_FLG_JIT_INTRINSIC)
+    {
+        // Do not report the get_IsSupported method as an intrinsic. This will turn the call into a regular
+        // call. We also make sure none of the hardware intrinsic method bodies get pregenerated in crossgen
+        // (see ZapInfo::CompileMethod) but get JITted instead. The JITted method will have the correct
+        // answer for the CPU the code is running on.
+
+        const char* namespaceName;
+        const char* methodName = pJitInfo->getMethodNameFromMetadata(ftn, nullptr, &namespaceName, nullptr);
+
+        if (strcmp(methodName, "get_IsSupported") == 0
+            && strcmp(namespaceName, "System.Runtime.Intrinsics.X86") == 0)
```

Comment:
I think we should make an exception for `Sse.IsSupported` and `Sse2.IsSupported`, since they should always return true for `_TARGET_XARCH_` (and therefore they should be passed along to the JIT as intrinsics).

Reply:
The exception here is for `COMPlus_EnableSSE=0` scenarios, which is something that needs to be rationalized when the AOT functionality is eventually expanded beyond just S.P.Corelib.

I think, likely, that we will need to say any AOT'd code will not respect the `COMPlus_` variable for static scenarios (but may, for dynamic dispatch ones) and that JIT'd code will always respect the variables.

Reply:
Right. I view those that disable the baseline codegen, as relevant only for test (the others, arguably, might be of possible use to someone who wants to ensure that they get the same code across platforms, I suppose).

Reply:
Added exception for Sse, Sse2, Sse.X64, Sse2.X64.